gnocchi/indexer/sqlalchemy.py

```diff
@@ -521,9 +524,9 @@ def fill_null_attribute_values(self, engine, name, rt, session,
                     f = QueryTransformer.build_filter(
```

Comment:
GitHub won't let me leave comments but I think the use of `str(url)` on line 301 above also needs to be updated. By default this will sanitise the URL and replace passwords with `***`. To avoid this, you need to use `url.render_as_string(hide_password=False)` instead. We encountered this in [nova also](https://review.opendev.org/c/openstack/nova/+/880669/4/nova/cmd/manage.py#761).

Reply:
Don't follow if you talk about the code in dress_url() or about __str__(), the code in __str__() could most definitely be cleaned with that approach but the other function is nothing exposed.

I will look into it and perhaps fix that in a separate PR.

Reply:
Oh reading it again, now I understand. I will check that.